src/Tribe/Aggregator/Service.php

```diff
@@ -161,8 +161,18 @@ public function build_url( $endpoint, $data = [] ) {
 			return $api;
 		}
 
-		// Build the URL
-		$url = "{$api->domain}{$api->path}{$api->version}/{$endpoint}";
+		/**
+		 * Allow to change the constructed URL used for EA.
+		 *
+		 * @since TBD
+		 *
+		 * @param string   $url      The completed URL generated.
+		 * @param string   $endpoint The path of the endpoint inside of the base url.
+		 * @param stdClass $api      An object representing the properties of the API.
+		 *
+		 * @return string The generated URL.
+		 */
+		$url = apply_filters( 'tribe_events_aggregator_build_url', "{$api->domain}{$api->path}{$api->version}/{$endpoint}/", $endpoint, $api );
```

Comment:
I wonder if this should be a method.

Reply:
I'm not sure if I get correctly, which part specifically the URL construction or the URL value?